api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxTbPrevCohortQueries.java

```diff
@@ -0,0 +1,193 @@
+package org.openmrs.module.eptsreports.reporting.library.cohorts;
+
+import java.util.Date;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.reporting.library.queries.TxTbPrevQueriesInterface;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.CompositionCohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.SqlCohortDefinition;
+import org.openmrs.module.reporting.definition.library.DocumentedDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TxTbPrevCohortQueries {
+
+  @Autowired private GenericCohortQueries genericCohorts;
+
+  @DocumentedDefinition(value = "findPatientWhoStartedTpi")
+  public CohortDefinition findPatientWhoStartedTpi() {
+    final SqlCohortDefinition definition = new SqlCohortDefinition();
+
+    definition.setName("findPatientWhoStartedTpi");
+    definition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    definition.addParameter(new Parameter("endDate", "End Date", Date.class));
+    definition.addParameter(new Parameter("location", "location", Location.class));
+
+    definition.setQuery(TxTbPrevQueriesInterface.QUERY.findPatientWhoStartedTpi);
+
+    return definition;
+  }
+
+  @DocumentedDefinition(value = "findPatientsWhoStartedTpi6MonthsAgoAndWhoEndedTpi")
+  public CohortDefinition findPatientsWhoStartedTpi6MonthsAgoAndWhoEndedTpi() {
+    final SqlCohortDefinition definition = new SqlCohortDefinition();
+
+    definition.setName("findPatientWhoStartedTpi");
+    definition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    definition.addParameter(new Parameter("endDate", "End Date", Date.class));
+    definition.addParameter(new Parameter("location", "location", Location.class));
+
+    definition.setQuery(
+        TxTbPrevQueriesInterface.QUERY.findPatientsWhoStartedTpi6MonthsAgoAndWhoEndedTpi);
+
+    return definition;
+  }
+
+  @DocumentedDefinition(value = "findPatientsTransferredOut")
+  public CohortDefinition findPatientsTransferredOut() {
+    final SqlCohortDefinition definition = new SqlCohortDefinition();
+
+    definition.setName("findPatientWhoStartedTpi");
+    definition.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    definition.addParameter(new Parameter("endDate", "End Date", Date.class));
+    definition.addParameter(new Parameter("location", "location", Location.class));
+
+    definition.setQuery(TxTbPrevQueriesInterface.QUERY.findPatientsTransferredOut);
+
+    return definition;
+  }
```

Comment:
@emaposse  I don't see the exclusions for patients whose after being marked as transferred out still pickingup or having  consultation.

Reply:
@Guimino  this part of the code solves that problem
https://github.com/FriendsInGlobalHealth/openmrs-module-eptsreports/blob/e5b69e92ff2c5da382fd630aa9673e109d77284b/api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxTbPrevQueriesInterface.java#L173-L177

Reply:
Ok Cool